File: ruby/lib/Sikaru/types/trace_metadata.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class TraceMetadata < Internal::Types::Model
      field :account_id, -> { String }, optional: false, nullable: false

      field :base_harness_version_id, -> { String }, optional: true, nullable: false

      field :compatibility_profile_id, -> { String }, optional: true, nullable: false

      field :conversation_id, -> { String }, optional: true, nullable: false

      field :converter_version, -> { String }, optional: true, nullable: false

      field :correlation_id, -> { String }, optional: true, nullable: false

      field :harness_id, -> { String }, optional: true, nullable: false

      field :harness_version_id, -> { String }, optional: true, nullable: false

      field :import_generation_id, -> { String }, optional: true, nullable: false

      field :job_id, -> { String }, optional: true, nullable: false

      field :organization_id, -> { String }, optional: true, nullable: false

      field :product_id, -> { String }, optional: true, nullable: false

      field :project_id, -> { String }, optional: false, nullable: false

      field :source, -> { String }, optional: false, nullable: false

      field :source_instance, -> { String }, optional: true, nullable: false

      field :tenant_id, -> { String }, optional: true, nullable: false

      field :trace_id, -> { String }, optional: true, nullable: false

      field :user_id, -> { String }, optional: true, nullable: false
    end
  end
end

File: python/src/sikaru_api/types/trace_metadata.py

import typing

import pydantic
from ..core.pydantic_utilities import IS_PYDANTIC_V2, UniversalBaseModel


class TraceMetadata(UniversalBaseModel):
    account_id: str
    base_harness_version_id: typing.Optional[str] = None
    compatibility_profile_id: typing.Optional[str] = None
    conversation_id: typing.Optional[str] = None
    converter_version: typing.Optional[str] = None
    correlation_id: typing.Optional[str] = None
    harness_id: typing.Optional[str] = None
    harness_version_id: typing.Optional[str] = None
    import_generation_id: typing.Optional[str] = None
    job_id: typing.Optional[str] = None
    organization_id: typing.Optional[str] = None
    product_id: typing.Optional[str] = None
    project_id: str
    source: str
    source_instance: typing.Optional[str] = None
    tenant_id: typing.Optional[str] = None
    trace_id: typing.Optional[str] = None
    user_id: typing.Optional[str] = None

    if IS_PYDANTIC_V2:
        model_config: typing.ClassVar[pydantic.ConfigDict] = pydantic.ConfigDict(extra="allow", frozen=True)  # type: ignore # Pydantic v2
    else:

        class Config:
            frozen = True
            smart_union = True
            extra = pydantic.Extra.allow
